### PR TITLE
Typed i18next

### PR DESCRIPTION
**Pull request by bhollis**
This utilizes the new types available in i18next to find mistakes in our usage of translation keys. I'm not actually sure the extra typechecking time is worth it, but I did find some mistakes. We could probably also just scan `en.json` for empty keys, though somehow that didn't cover all of them.

### CODE REVIEW

File: src/app/destiny1/activities/Activities.tsx [hunk 1]
```diff
@@ -79,7 +79,7 @@ export default function Activities({ account }: Props) {
       tier.activityData.recommendedLight === 390
         ? '390'
         : tier.tierDisplayName
-        ? t(`Activities.${tier.tierDisplayName}`, {
+        ? t(`Activities.${tier.tierDisplayName}` as I18nKey, {
```
Comment (bhollis) [inline src/app/destiny1/activities/Activities.tsx:82]: I'm not sure I know a better way to handle these. Perhaps if things like `tierDisplayName` were modeled as string union types?

Comment (robojumper) [reply src/app/destiny1/activities/Activities.tsx:82]: The TS type checker can certainly handle that functionality by distributing the interpolation across the string union.

> extra typechecking time

I was thinking "how bad can it be?" but when playing around with it this added an extra 6s of latency to some IDE hover type hints, which is pretty extreme. I wonder if this is expected or just some unoptimized pathological case in the TS type checker...

Comment (robojumper) [reply src/app/destiny1/activities/Activities.tsx:82]: Looking at https://github.com/i18next/i18next/issues, parts of this feature seem to be really heavy on the TS type checker and cause a variety of errors (OOMs, type instantiation recursion limits, ...) in some projects.

Comment (bhollis) [reply src/app/destiny1/activities/Activities.tsx:82]: Yeah, they've been working on it for several releases. This is the first time it's actually worked.

File: src/app/utils/dim-error.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ export class DimError extends Error {
   cause?: Error;
 
   /** Pass in just a message key to set the message to the localized version of that key, or override with the second parameter. */
-  constructor(messageKey: string, message?: string) {
+  constructor(messageKey: I18nKey, message?: string) {
```
Comment (robojumper) [inline src/app/utils/dim-error.ts:17]: I feel like this should be something like this as to not require a non-localized code to be present in the localization files. 

```suggestion
  /** Pass in just a message key to set the message to the localized version of that key */
  constructor(messageKey: I18nKey);
  /** Use the key as an internal code and provide a localized message instead */
  constructor(messageKey: string, message: string);

  constructor(messageKey: I18nKey, message?: string) {
```

**Pull request closed without merging**